### PR TITLE
Add ways to turn P2 into P3 #326

### PR DESCRIPTION
**Pull request by janbols**
Added append methods to Px classes for #326

### CODE REVIEW

File: core/src/main/java/fj/P2.java [hunk 1]
```diff
@@ -1,12 +1,13 @@
 package fj;
 
-import static fj.Function.*;
```
Comment (jbgi) [inline core/src/main/java/fj/P2.java]: Can you restore the import order as before?

Comment (janbols) [reply core/src/main/java/fj/P2.java]: Ok, just did.

**Comment by gliptak** [comment]
Would we like to see some (basic) tests against this?

**Comment by jbgi** [comment]
@gliptak I think the signature of the methods already enforce a unique possible implementation, so tests are not necessary in this case.

**Comment by jbgi** [comment]
Thanks!

**Pull request merged**